packages/api/src/auth/auth.service.ts

```diff
@@ -0,0 +1,32 @@
+import { ConflictException, Injectable } from '@nestjs/common';
+import { EventBus } from '@nestjs/cqrs';
+import { Prisma } from '@prisma/client';
+
+import { registerError } from '@coderscamp/shared/models/auth/register';
+import { pick } from '@coderscamp/shared/utils/object';
+
+import { AuthRepository } from './auth.repository';
+import { UserRegistrationCompletedEvent } from './events/user-registration-completed.event';
+import { hashPassword } from './local/local.utils';
+
+@Injectable()
+export class AuthService {
+  constructor(private readonly eventBus: EventBus, private readonly authRepository: AuthRepository) {}
+
+  async register(data: Prisma.UserRegistrationCreateInput & Prisma.AuthUserCreateInput) {
+    const current = await this.authRepository.findUserRegistration({ where: { email: data.email } });
+
+    if (current) {
+      throw new ConflictException(registerError.REGISTRATION_FORM_ALREADY_EXISTS);
+    }
+
+    const userRegistrationData = pick(data, ['fullName', 'email']);
+    const password = await hashPassword(data.password);
+
+    const { id } = await this.authRepository.createUserRegistration({ data: userRegistrationData });
+
```

Comment:
What's the difference between userRegistration and authUser? Do we need them both as separate entities?

Reply:
You're right that `userRegistration` has no much use for now. I created it as we already know it will be needed when authentication flow expands by other steps (e.g. email confirmation #185). In such a case you will treat `userRegistration` as a bucket that stores data about the registration process until it finishes.

When it comes to `authUser` it's a table used by the entire auth module to perform auth-related operations. Doing it this way allows us to keep Auth module independent from any other modules.

Reply:
@KonradSzwarc - beautiful :)